### ISSUE TITLE
redundant_closure should detect trivial method calls

### ISSUE THREAD

**Issue by asomers** [rust-lang/rust-clippy#3469]
The `redundant_closure` lint currently works for functions, but not for methods.  It should also detect redundant closures like this: `Some(x).map(x.foo())`.

```
> cargo +nightly clippy -V
clippy 0.0.212 (481f7880 2018-10-30)
``
